src/service-worker.ts

```diff
@@ -98,97 +86,76 @@ export async function generateServiceWorkerConfig(
-    // entrypoint might not be called "index.html", and 2) this case is already
-    // handled by the navigateFallback configuration above. Simplest to just
-    // disable this feature.
-    swPrecacheConfig.directoryIndex = '';
-  }
-
-  // swPrecache will determine the right urls by stripping buildRoot.
-  // NOTE:(usergenic) sw-precache generate() apparently replaces the
-  // prefix on an already posixified version of the path on win32.
-  //
-  // We include a trailing slash in `stripPrefix` so that we remove leading
-  // slashes on the pre-cache asset URLs, hence producing relative URLs
-  // instead of absolute. We want relative URLs for builds mounted at non-root
-  // paths. Note that service worker fetches are relative to its own URL.
-  swPrecacheConfig.stripPrefix = addTrailingSlash(posixifyPath(buildRoot));
-
   if (options.basePath) {
-    // TODO Drop this feature once CLI doesn't depend on it.
-    let replacePrefix = options.basePath;
-    if (!replacePrefix.endsWith('/')) {
-      replacePrefix = replacePrefix + '/';
-    }
-    if (swPrecacheConfig.replacePrefix) {
-      console.info(
-          `Replacing service worker configuration's ` +
-          `replacePrefix option (${swPrecacheConfig.replacePrefix}) ` +
-          `with the build configuration's basePath (${replacePrefix}).`);
-    }
-    swPrecacheConfig.replacePrefix = replacePrefix;
+    workboxConfig.manifestTransforms = [(entries: ManifestEntry[]) => {
+      entries.forEach(entry => {
+        entry.url = path.join(options.basePath, entry.url);
+      });
+      return {
+        manifest: entries
+      }
+    }];
   }
 
   // static files will be pre-cached
-  swPrecacheConfig.staticFileGlobs = staticFileGlobs;
+  workboxConfig.globPatterns = staticFileGlobs;
 
-  // Log service-worker helpful output at the debug log level
-  swPrecacheConfig.logger = swPrecacheConfig.logger || logger.debug;
-
-  return swPrecacheConfig;
+  return workboxConfig;
 }
 
 /**
  * Returns a promise that resolves with a generated service worker (the file
  * contents), based off of the options provided.
  */
 export async function generateServiceWorker(options: AddServiceWorkerOptions):
-    Promise<Buffer> {
-  const swPrecacheConfig = await generateServiceWorkerConfig(options);
-  return await<Promise<Buffer>>(new Promise((resolve, reject) => {
-    logger.debug(`writing service worker...`, swPrecacheConfig);
-    swPrecacheGenerate(
-        swPrecacheConfig, (err?: Error, fileContents?: string) => {
-          if (err) {
-            reject(err);
-          } else {
-            resolve(new Buffer(fileContents));
-          }
-        });
+  Promise<Buffer> {
+  const workboxConfig = await generateServiceWorkerConfig(options);
+  return await <Promise<Buffer>>(new Promise((resolve) => {
+    logger.debug(`writing service worker...`, workboxConfig);
+    generateSWString(workboxConfig).then(({ swString }) => {
```

Comment:
I think it's worth checking for `warnings` (which is an array of strings) in the `generateSWString()` response, and logging anything that's returned.

Reply:
Done